src/Microsoft.AspNet.Hosting/Program.cs

```diff
@@ -29,7 +29,9 @@ public void Main(string[] args)
             var tempConfig = tempBuilder.Build();
             var configFilePath = tempConfig[ConfigFileKey] ?? HostingJsonFile;
 
-            var appBasePath = _serviceProvider.GetRequiredService<IApplicationEnvironment>().ApplicationBasePath;
+            var appEnv = _serviceProvider.GetRequiredService<IApplicationEnvironment>();
+            appEnv.SetData("CommandLineArguments",args);
```

Comment:
I don't like it. If we were going to expose the command line arguments passed into hosting, it would need to be something a bit more first class. Using the SetData, GetData is too much of a hack IMO.